step-controller-plugins/step-controller-plugins-timeseries/src/main/java/step/plugins/timeseries/TimeSeriesHandler.java

```diff
@@ -272,7 +272,9 @@ private TimeSeriesAggregationQuery mapToQuery(FetchBucketsRequest request, TimeS
                                 TimeSeriesFilterBuilder.buildFilter(request.getOqlFilter()),
                                 TimeSeriesFilterBuilder.buildFilter(request.getParams()))
                 ))
-                .withGroupDimensions(request.getGroupDimensions());
+                .withGroupDimensions(request.getGroupDimensions())
+                .withCollectAttributes(request.getCollectAttributeKeys(),
```

Comment:
Is request.getCollectAttributeKeys() always set? i.e. is it set to null or to an empty collection when attribute collection is not needed? If not needed we should ensure that the collection is null (which improves the performance of BucketBuilder.accumulateAttributes) and it would be in this case preferable not to call withCollectAttributes at all.

Reply:
Changed as requested; removed the check of "isEmpty" in BucketBuilder